### PR TITLE
Remove unnecessary stuff in boosted-*.scss files

### LINKED ISSUE #2153: `boosted-grid.css` contains too much stuff
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

`boosted-grid.css` seems to embed too much CSS. For example, it contains `.text-*` classes that shouldn't be in a grid CSS file:
https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/235a887808b7825a135b470fe99e3ade531b470b/dist/css/boosted-grid.css#L2459

This issue requires the following analysis and actions (non-exhaustive):
* Compare `boosted-grid.css` and `bootstrap-grid.css` to check what's in our file that shouldn't be
* Fix the issue for `boosted-grid.css`
* Do the same thing for all `boosted-*.css` files (except of course `boosted.css`) while we're at it

### Reduced test cases

Check directly the files in the `dist` repository.

### What operating system(s) are you seeing the problem on?

Windows, macOS, Android, iOS, Linux

### What browser(s) are you seeing the problem on?

Chrome, Safari, Firefox, Microsoft Edge, Opera

### What version of Boosted are you using?

v5.2.3 / v5.3.0

### PR DESCRIPTION
**Pull request by MewenLeHo**
### Related issues

Closes #2153 

### Description

Remove unnecessary stuff in boosted-grid.scss

### Motivation & Context

Lighter code and smaller bundle size.

### Types of change

- Bug fix (non-breaking which fixes an issue)

### Live previews

- <https://deploy-preview-2196--boosted.netlify.app/>

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

#### Accessibility

- (NA) My change follows accessibility good practices; I have at least run axe

#### Design

- (NA) My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
- (NA) My change is compatible with a responsive display

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
- (NA) I have added JavaScript unit tests to cover my changes
- (NA) I have added SCSS unit tests to cover my changes

#### Documentation

- (NA) My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- [ ] My change introduces changes to the migration guide
- (NA) My new component is well displayed in [Storybook](https://deploy-preview-2196--boosted.netlify.app/storybook)
- (NA) My new component is compatible with RTL
- (NA) Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
- (NA) Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
- [ ] Code review
- [ ] Design review
- [ ] A11y review

#### After the merge

- [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

**Comment by MewenLeHo** [comment]
IMY we still have a problem because `running npm run` dist with this configuration removes some necessary stuff too 🤔

File: scss/boosted-grid.scss
Comment (MewenLeHo) [inline scss/boosted-grid.scss:74]: A PR was made on Bootstrap's side for this part.
Currently waiting for their answer...

**Review by louismaximepiton — APPROVED** [review]
LGTM, great to have this!

File: scss/boosted-grid.scss
Comment (MewenLeHo) [reply scss/boosted-grid.scss:17]: Not mandatory but it seems more logical to me 🤔

File: scss/boosted-grid.scss
Comment (julien-deramond) [reply scss/boosted-grid.scss:74]: See https://github.com/twbs/bootstrap/pull/39068

Comment (julien-deramond) [reply scss/boosted-grid.scss:74]: Can I drop this modification so that we can merge this PR? And if it's accepted upstream, it'll be integrated when the Bootstrap commits will be merged into Boosted?

**Review by julien-deramond — APPROVED** [review]
LGTM, good job!

**Pull request merged**
